ddpui/services/dashboard_service.py

```diff
@@ -30,9 +39,575 @@
+    name: Optional[str] = None
+    settings: Optional[dict] = None
+    order: int = 0
+
+
+@dataclass
+class LockInfo:
+    """Data class for lock information"""
+
+    lock_token: str
+    expires_at: datetime
+    locked_by_email: str
+
+
 class DashboardService:
     """Service class for dashboard-related operations"""
 
+    # =========================================================================
+    # CRUD Operations
+    # =========================================================================
+
+    @staticmethod
+    def get_dashboard(dashboard_id: int, org: Org) -> Dashboard:
+        """Get a dashboard by ID for an organization.
+
+        Args:
+            dashboard_id: The dashboard ID
+            org: The organization
+
+        Returns:
+            Dashboard instance
+
+        Raises:
+            DashboardNotFoundError: If dashboard doesn't exist or doesn't belong to org
+        """
+        try:
+            return Dashboard.objects.prefetch_related("filters").get(id=dashboard_id, org=org)
+        except Dashboard.DoesNotExist:
+            raise DashboardNotFoundError(dashboard_id)
+
+    @staticmethod
+    def list_dashboards(
+        org: Org,
+        dashboard_type: Optional[str] = None,
+        search: Optional[str] = None,
+        is_published: Optional[bool] = None,
+    ) -> List[Dashboard]:
+        """List dashboards for an organization with filtering.
+
+        Args:
+            org: The organization
+            dashboard_type: Optional filter by dashboard type
+            search: Optional search term for title/description
+            is_published: Optional filter by published status
+
+        Returns:
+            List of Dashboard instances
+        """
+        query = Q(org=org)
+
+        if dashboard_type:
+            query &= Q(dashboard_type=dashboard_type)
+
+        if search:
+            query &= Q(title__icontains=search) | Q(description__icontains=search)
+
+        if is_published is not None:
+            query &= Q(is_published=is_published)
+
+        return list(Dashboard.objects.filter(query).order_by("-updated_at"))
+
+    @staticmethod
+    def create_dashboard(data: DashboardData, orguser: OrgUser) -> Dashboard:
+        """Create a new dashboard.
+
+        Args:
+            data: Dashboard creation data
+            orguser: The user creating the dashboard
+
+        Returns:
+            Created Dashboard instance
+        """
+        dashboard = Dashboard.objects.create(
+            title=data.title,
+            description=data.description,
+            grid_columns=data.grid_columns,
+            created_by=orguser,
+            org=orguser.org,
+            last_modified_by=orguser,
+        )
+
+        logger.info(f"Created dashboard {dashboard.id} for org {orguser.org.id}")
+        return dashboard
+
+    @staticmethod
+    def update_dashboard(
+        dashboard_id: int,
+        org: Org,
+        orguser: OrgUser,
+        title: Optional[str] = None,
```

Comment:
any reason why we aren't using the schema `DashboardUpdate` as an input in `update_dashboard` ?

same for other apis. we dont want to type the same payload as arguments if its already available.

thoughts ?